server/repository/userRepository.js

```diff
@@ -21,19 +21,67 @@ class UserRepository {
   /**
  * Finds a user by email
  * @param {string} email Email to search by
+ * @param {string} username username
  * @returns {object | null} User object or null if user is not found
  */
-  static async getUserByEmail(email) {
-    const user = await User.findOne({
-      where: { email },
+  static async getUserByCredentials(email, username) {
```

Comment:
Why was this change required?

Reply:
We decided to check both email and username when a user is signing up. This is because we don't want two users to have the same username. Or a user to sign up again using the same username and/or email.